### PR TITLE
add tax calculation mode to our SDK

### LINKED ISSUE #1603: add documentation for taxCalculationMode
we should add support for our backend ticket [#719](https://github.com/sphereio/dev.commercetools.com/pull/719)

### PR DESCRIPTION
**Pull request by acbeni**
Closes #1603

### CODE REVIEW

File: commercetools-models/src/test/java/io/sphere/sdk/client/UserAgentIntegrationTest.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ public void userAgent() throws Exception {
         }
     }
 
-    private static class FakeHttpClient implements HttpClient {
+    public static class FakeHttpClient implements HttpClient {
```
Comment (katmatt) [inline commercetools-models/src/test/java/io/sphere/sdk/client/UserAgentIntegrationTest.java]: Is this really related to this issue?

File: commercetools-models/src/main/java/io/sphere/sdk/carts/TaxCalculationMode.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package io.sphere.sdk.carts;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import io.sphere.sdk.models.SphereEnumeration;
+
+/**
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/carts/TaxCalculationMode.java]: I really like that you added this javadoc 👍

File: commercetools-models/src/test/java/io/sphere/sdk/client/UserAgentIntegrationTest.java (hunk 1 above)
Comment (acbeni) [reply commercetools-models/src/test/java/io/sphere/sdk/client/UserAgentIntegrationTest.java]: No but it caused some access problems with OSGi so i decided to make it public, just as a shortcut to fix the problem.

Comment (katmatt) [reply commercetools-models/src/test/java/io/sphere/sdk/client/UserAgentIntegrationTest.java]: Ok and thanks for your explanation 👍
